src/eth_provider/provider.rs

```diff
@@ -128,6 +128,8 @@ pub trait EthereumProvider {
         &self,
         block_id: Option<BlockId>,
     ) -> EthProviderResult<Option<Vec<reth_rpc_types::Transaction>>>;
+    /// Returns a vec of pending pool transactions.
+    async fn txpool_transactions(&self) -> EthProviderResult<Vec<StoredPendingTransaction>>;
```

Comment:
I'd prefer we avoid introducing our types in the trait, keep it reth only

Reply:
Agreed, just fixed